docs/tutorial/dark-mode.md

```diff
@@ -0,0 +1,183 @@
+the system's dark mode setting. You can do this by using the
+[prefer-color-scheme] CSS @media query.
+
+## Manually update your own interfaces
+
+If you want to manually switch between light/dark modes, you can do this by
+setting the desired mode in the
+[themeSource](https://www.electronjs.org/docs/api/native-theme#nativethemethemesource)
+property of the `nativeTheme` module.
+
+## Example
+
+We'll start with a working application from the
+[Quick Start Guide](quick-start.md) and add functionality gradually.
+
+First, let's add a page where a user can toggle between light and dark modes.
+By default, Electron will follow the system's dark mode preference, so we
+initially set the theme source as "System".
+
+Add the following lines to the `index.html` file:
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8">
+    <title>Hello World!</title>
+    <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'unsafe-inline';" />
+    <link rel="stylesheet" type="text/css" href="./styles.css">
+</head>
+<body>
+    <h1>Hello World!</h1>
+    <p>Current theme source: <strong id="theme-source">System</strong></p>
+
+    <button id="toggle-dark-mode">Toggle Dark Mode</button>
+    <button id="reset-to-system">Reset to System Theme</button>
+
+    <script src="renderer.js"></script>
+  </body>
+</body>
+</html>
+```
+
+Next, add event listeners to the toggle buttons and response handlers. The
+listeners will listen for click events and update the content on the web page
+accordingly:
+
+* when the "Toggle Dark Mode" button is clicked, Electron sends the
+`dark-mode:toggle` message (event) to the Main process that triggers theme
+change, and sets the Current theme source based on the response from the
+Main process.
+* when the "Reset to System Theme" button is clicked, Electron sends
+the `dark-mode:system` message (event) to the Main process that tells that
+the theme should be set to default and sets the Current theme source to
+`System`.
+
+To ddd listeners and handlers, add the following lines to the `renderer.js` file:
+
+```js
+const { ipcRenderer } = require('electron')
+
+document.getElementById('toggle-dark-mode').addEventListener('click', async () => {
+  const isDarkMode = await ipcRenderer.invoke('dark-mode:toggle')
+  document.getElementById('theme-source').innerHTML = isDarkMode ? 'Dark' : 'Light'
+})
+
+document.getElementById('reset-to-system').addEventListener('click', async () => {
+  await ipcRenderer.invoke('dark-mode:system')
+  document.getElementById('theme-source').innerHTML = 'System'
+})
+```
+
+The next step is to update the `main.js` file to handle events from the
+Renderer process. Depending on the received event (`dark-mode:toggle` or
+`dark-mode:system`), we update the [`nativeTheme.themeSource`](../api/native-theme.md#nativethemethemesource)
+API to apply the desired theme on the system's native UI elements (e.g. context
+menus on macOS).
```

Comment:
```suggestion

If you run your code at this point, you'll see that your buttons don't do anything
just yet, and your Main process will output an error like this when you click on your
buttons: 
`Error occurred in handler for 'dark-mode:toggle': No handler registered for 'dark-mode:toggle'`
This is normal — we haven't actually touched any `nativeTheme` code yet.

Now that we're done wiring the IPC from the Renderer's side, the next step
is to update the `main.js` file to handle events from the Renderer process.

Depending on the received event, we update the [`nativeTheme.themeSource`](../api/native-theme.md#nativethemethemesource)
property to apply the desired theme on the system's native UI elements (e.g. context
menus) and propagate the preferred color scheme to the Renderer process:
* Upon receiving `dark-mode:toggle`, we check if the dark theme is currently
active using the `nativeTheme.shouldUseDarkColors` property, and set the
`themeSource` to the opposite theme.
* Upon receiving `dark-mode:system`, we reset the `themeSource` to `system`.
```

A few copy-editing suggestions:

* Added a note that the code doesn't work yet after the Renderer step, and expected behaviour at that point.
* Added bullet points explaining each handler to mirror what we had in the last step.